adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttProtocolAdapterProperties.java

```diff
@@ -49,13 +66,49 @@ public final int getCommandAckTimeout() {
      * <p>
      * The default value is {@link #DEFAULT_COMMAND_ACK_TIMEOUT}.
      *
+     * @deprecated Use {@link #setSendMessageToDeviceTimeout(long)} ()} instead.
+     *
      * @param timeout The time out in milliseconds.
      * @throws IllegalArgumentException if the timeout is negative.
      */
+    @Deprecated(forRemoval = true)
     public final void setCommandAckTimeout(final int timeout) {
         if (timeout < 0) {
             throw new IllegalArgumentException("timeout must not be negative");
         }
         this.commandAckTimeout = timeout;
     }
+
+    /**
+     * Gets the waiting for acknowledgement time out in milliseconds for commands published with QoS 1.
+     * <p>
+     * This time out is used by the MQTT adapter for commands published with QoS 1. If there is no acknowledgement
+     * within this time limit, then the command is settled with the <em>released</em> outcome.
+     * <p>
+     * The default value of this property is {@link ClientConfigProperties#getSendMessageTimeout()}.
+     *
+     * @return The time out in milliseconds.
+     */
+    public long getSendMessageToDeviceTimeout() {
+        return sendMessageToDeviceTimeout;
+    }
```

Comment:
This method doesn't seem to be used. I would have expected it to be used in the place that up til now `getCommandAckTimeout()` was used in (i.e. in `CommandSubscriptionsManager.startTimer`).
For backwards compatibility, the `commandAckTimeout` value should still be used if `sendMessageToDeviceTimeout` isn't set.

But then there is a conflict with using `ClientConfigProperties#getSendMessageTimeout()` as default for `sendMessageToDeviceTimeout`. That means that assigning the default value of `sendMessageToDeviceTimeout` in the constructor would have to be changed (in favor of applying the order of precedence of the 3 possible values in the `getSendMessageToDeviceTimeout` method above).

But, I now think that using the `sendMessageTimeout` value of the _messaging_ configuration properties as default value might be too confusing here after all. It's also the first time that we have properties of different endpoints influence one another. And also from the constructor signature it doesn't become clear, which actual `ClientConfigProperties` are to be used here.

I think, for the time being (until we have an extra set of some default properties), we should just use the `1 second` actual default value of `sendMessageTimeout` as default for `sendMessageToDeviceTimeout`, not using the _messaging_ client properties here. WDYT?

Reply:
Well, actually I found it confusing when I implemented it but I blamed it on me working on my first issue on Hono.

Just to make sure I got you right: we would remove the reference to `ClientConfigProperties` completely then from the constructor arguments (meaning not only in the MQTT adapter but as well in the AMQP adapter)?

Reply:
Yes, and in the `KuraAdapterProperties`.

Reply:
Okay, I reverted my changes.